include/pmacc/eventSystem/tasks/TaskReceiveMPI.hpp

```diff
@@ -46,12 +46,29 @@ class TaskReceiveMPI : public MPITask
 
     virtual void init()
     {
+
+        Buffer<TYPE, DIM>* dst = nullptr;
+
+        if(Environment<>::get().isMpiDirectEnabled())
+        {
+            if(exchange->hasDeviceDoubleBuffer())
+                dst = &(exchange->getDeviceDoubleBuffer());
+            else
+                dst = &(exchange->getDeviceBuffer());
+        }
+        else
+        {
+            dst = &(exchange->getHostBuffer());
+        }
```

Comment:
This logic can be moved into a method of exchange or a free function to return a `Buffer<TYPE, DIM>*`.